docs/design/proposals/destructive-operations/destructive-operations.md

```diff
@@ -0,0 +1,130 @@
+
+## AWS Removal Policy
+
+### Requirements
+
+1. Multiple levels of granularity over resource protection (controller-wide, namespace-wide and resource-wide)
+2. Protection should be defined per-service (if namespace-wide)
+3. Users are able to define either of:
+    1. The AWS resource should be deleted along with the CR
+    2. The AWS resource is left in-tact after the CR is deleted
+
+### Investigation
+
+A number of K8s resources already have similar functionality:
+
+* Helm uses an annotation: `helm.sh/resource-policy: keep`
+* Google Config Connector uses an annotation: `cnrm.cloud.google.com/deletion-policy: abandon`
+* Skaffold uses a label: `skaffold.dev/cleanup: "false"`
+* K8s PV use a spec field: `persistentVolumeReclaimPolicy``: Retain`
+
+
+Existing AWS products have their own variants:
+
+* AWS CDK uses a spec field: `removalPolicy: RemovalPolicy.RETAIN`
+* CloudFormation uses a metadata field on the resource: `DeletionPolicy: Retain`
+
+CloudFormation and CDK both provide the ability to snapshot given resources before deleting them (mostly databases). While this is outside of the scope for these requirements, it’s not outside the scope of ACK in the future.
+
+## Proposed Option
+
+The following proposed option will introduce:
+
+* Two new annotations, one service-wide and one ACK-wide, to be applied on namespace and ACK resources
+* One new optional command line argument for the controller binary
+
+(Requirements 1) Following the current CARM design, which also requires the same levels of granularity, it is consistent to mark the policies in the annotation metadata for the namespace and resources, and to create a new command line argument on the controller for the controller-wide policies. The controller would place precedence on the resource policy, then the namespace policy and finally the controller policy - the annotation on the object with the highest precedence will set the policy for the given resource.
+
+(Requirements 2) The full annotation path needs to consist of an annotation prefix (typically a URL) before the annotation name. One variation should concern every ACK controller regardless of service, and one should discriminate for controllers only concerned with a single service. Cross-service annotations, finalizers and CRDS are already configured with the `services.k8s.aws/` prefix. For this option, prepending the service identifier to the beginning would identify that the annotation applies only to the given service, eg. `s3.services.k8s.aws/` . Hence, there would be two prefixes:
+
+* `services.k8s.aws/`: All ACK controllers will read the policy of this namespace/resource
+* `<service>.services.k8s.aws/`: Only the <service> ACK controller will read the policy of this namespace (taking precedence over the former option)
```

Comment:
Since there will always be only one controller responsible of deleting specific resources. Why do we need the `<service>.services.k8s.aws/` prefixed annotation?

Reply:
NVM looks like this is an namespace level annotation ++